ggshield/verticals/auth/oauth.py

```
@@ -163,19 +158,14 @@ def _redirect_to_login(self) -> None:
-            except OSError:
-                continue
-        else:
-            raise UnexpectedError("Could not find unoccupied port.")
+        try:
+            self.server = HTTPServer(
+                ("127.0.0.1", 0),  # Let OS choose an available port
+                functools.partial(RequestHandler, self),
+            )
+            self._port = self.server.server_port  # Get the assigned port
+        except OSError as e:
+            raise UnexpectedError(f"Could not create local server: {e}")
```

Ruff (0.14.4)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)